jaeger_client/reporter.py

```diff
@@ -186,44 +178,22 @@ def _consume_queue(self):
                         # don't return yet, submit accumulated spans first
                         break
                     else:
-                        spans.append(span)
-            if spans:
-                yield self._submit(spans)
-                for _ in spans:
+                        self._sender.append(span)
+
+            if self._sender.span_count:
+                num_spans = self._sender.span_count
+                exc, error_msg = yield self._sender.flush()
```

Comment:
I don't get the benefit of that, honestly. What you are doing now is a `try ...except Exception ...` in the `Sender.flush()`, returning the exception plus a message.  Which is the same as an `raise AnyException("message")`

In the `Reporter` you just check if any kind of `exc` is raised, and the use the message returned. 

This is the very same of:
```python
try:
    yield self._sender.flush()
except Exception as exc:
   self.metrics.reporter_failure(num_spans)
   self.error_reporter.error(error_msg, exc)
else:
    self.metrics.reporter_success(num_spans)
```
but this is the most common and standard way to deal with exceptions, in Python.

The point is to keep the code "idiomatic".
In other languages, as Go, the standard is to have the pattern you are doing: multiple returns, one for the error/ok-boolean and the other for the "real" value.

Reply:
I was writing in this little github editor so I lost the context
```python
try:
    yield self._sender.flush()
except Exception as exc:
   self.metrics.reporter_failure(num_spans)
   self.error_reporter.error("'Failed to submit traces to jaeger-agent socket: {}".format(exc):, exc)
else:
    self.metrics.reporter_success(num_spans)
```

You can still catch below, in the sender, the specific error and provide a relevant message, carried with the exception itself.

Reply:
Providing a custom message in `send()` seems effective.  Are you comfortable with using something like `six.reraise` to add a custom exception with the traceback from `sys.exc_info()`?

Reply:
I would not add six as dependency just for that. And probably jaeger maitaners wouldn't too

Reply:
It already is one via thrift.

Reply:
Oh, didn't know it was included in thrift. Then I suppose it's allright to use it